src/middleware/favicon.rs

```diff
@@ -0,0 +1,46 @@
+use std::path::Path;
+
+use super::Middleware;
+use crate::get;
+use crate::response::File;
+
+/// Serves the favicon at the provided path argument.
+///
+/// # Example
+///
+/// ```
+/// use via::middleware::favicon;
+/// use via::{Next, Request};
+///
+/// type Error = Box<dyn std::error::Error + Send + Sync>;
+///
+/// #[tokio::main(flavor = "current_thread")]
+/// async fn main() -> Result<(), Error> {
+///     let mut app = via::app(());
+///
+///     app.at("/favicon.ico").respond(favicon("./favicon.ico"));
+///     Ok(())
+/// }
+/// ```
+///
+pub fn favicon<T>(path: impl AsRef<Path>) -> impl Middleware<T> {
```

Comment:
I'm considering removing this middleware in favor of using the following in an example or guide somewhere.

```rust
app.at("/favicon.ico").respond(via::get(|_, _| {
    File::open(PathBuf::from(PUBLIC_DIR).join("favicon.png"))
        .content_type("image/x-icon".to_owned())
        .with_last_modified()
        .serve()
}));
```